### PR TITLE
E2e/feature/ri 3596 add search capabilities to browser

### PR DESCRIPTION
**Pull request by vlad-dargel**

### CODE REVIEW

File: tests/e2e/helpers/keys.ts [hunk 1]
```diff
@@ -214,3 +214,13 @@ export async function deleteAllKeysFromDB(host: string, port: string): Promise<v
...
     });
 }
+
+/**
+* Verifying if the Keys are in the List of keys
+* @param keyNames The names of the keys
+*/
+export async function verifyKeysDisplayedInTheList(keyNames: string[]): Promise<void> {
+    for (const keyName of keyNames) {
+        await t.expect(await browserPage.isKeyIsDisplayedInTheList(keyName)).ok(`The key ${keyName} not found`);
```
Comment (ofersteinberg) [inline tests/e2e/helpers/keys.ts:224]: Consider using `.visible`

File: tests/e2e/pageObjects/browser-page.ts [hunk 2]
```diff
@@ -972,6 +980,17 @@ export class BrowserPage {
             await t.expect(scannedResults).gt(rememberedScanResults, { timeout: 3000 });
         }
     }
+
+    /**
+     * Open Select Index droprown and select option
+     * @param index The name of format
+     */
+    async selectIndexByName(index: string): Promise<void> {
+        const option = Selector(`[data-test-subj="mode-option-type-${index}"]`);
```
Comment (ofersteinberg) [inline tests/e2e/pageObjects/browser-page.ts:989]: I think the `"` are redundant

File: tests/e2e/tests/critical-path/browser/search-capabilities.e2e.ts [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+        // Verify that user see the "Select an index" message when he switch to Search
+        await t.click(browserPage.redisearchModeBtn);
+        await t.expect(browserPage.keyListTable.textContent).contains(notSelectedIndexText, 'Select an index message not displayed');
+
+        // Verify that user can search by index in Browser view
+        await browserPage.selectIndexByName(indexName);
+        await verifyKeysDisplayedInTheList(keyNames);
+        await t.expect((await browserPage.getKeySelectorByName(keyName)).exists).notOk('Key without index displayed after search');
+        // Verify that user can search by index plus key value
+        await browserPage.searchByKeyName('Hall School');
```
Comment (AlenaNaboka) [inline tests/e2e/tests/critical-path/browser/search-capabilities.e2e.ts:67]: You create am index with different indexed fields by Text, Numeric, Tag. 
Can you please add to test more complex check like search not only by key value but for description + name + type + city or something else?

File: tests/e2e/pageObjects/browser-page.ts (hunk 2 above)
Comment (vlad-dargel) [reply tests/e2e/pageObjects/browser-page.ts:989]: no, without `"` it doesn't work, I checked

File: tests/e2e/helpers/keys.ts (hunk 1 above)
Comment (vlad-dargel) [reply tests/e2e/helpers/keys.ts:224]: there is .exists check into the isKeyIsDisplayedInTheList() method
image

File: tests/e2e/tests/critical-path/browser/search-capabilities.e2e.ts (hunk 3 above)
Comment (vlad-dargel) [reply tests/e2e/tests/critical-path/browser/search-capabilities.e2e.ts:67]: thanks, added

**Review by AlenaNaboka — APPROVED** [review]

**Pull request merged**